File: pset6/credit/credit.py
num = int(input("Número: "))

pares = impares = total = contador = dezena = check = 0

i = 1

while num > 0:
    if i % 2 == 0:
        if ((num % 10) * 2) < 9:
            pares += (num % 10) * 2
        else:
            num2 = (num % 10) * 2
            num3 = num4 = 0

            while num2 > 0:
                num3 = (num2 % 10)
                num4 += num3
                num2 = int(num2 / 10)

            pares += num4
    else:
        impares += num % 10

    contador = i
    i += 1

    num = int(num / 10)

    if (num < 100) and (num > 10):
        dezena = num

total = pares + impares

check = total % 10

if check == 0 and contador == 15 and (dezena == 34 or dezena == 37):
    print("AMEX")
elif check == 0 and contador == 16 and (dezena >= 51 and dezena <= 55):
    print("MASTERCARD")
elif check == 0 and (contador == 13 or contador == 16) and (int(dezena / 10) == 4):
    print("VISA")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long numero = get_long("Número: ");
    int pares = 0;
    int impares = 0;
    int total = 0;
    int contador = 0;
    int dezena = 0;
    int check = 0;

    for (int i = 1; numero > 0; i++)
    {
        if (i % 2 == 0)
        {
            if (((numero % 10) * 2) < 9)
            {
                pares += ((numero % 10) * 2);
            }
            else
            {
                int num2 = ((numero % 10) * 2);
                int num3, num4 = 0;

                while (num2 > 0)
                {
                    num3 = (num2 % 10);
                    num4 += num3;
                    num2 = num2 / 10;
                }

                pares += num4;
            }

        }
        else
        {
            impares += (numero % 10);
        }

        contador = i;

        numero = numero / 10;

        if ((numero < 100) && (numero > 10))
        {
            dezena = numero;
        }
    }

    total = pares + impares;

    check = total % 10;

    if (check == 0 && contador == 15 && (dezena == 34 || dezena == 37))
    {
        printf("AMEX\n");
    }
    else if (check == 0 && contador == 16 && (dezena >= 51 && dezena <= 55))
    {
        printf("MASTERCARD\n");
    }
    else if (check == 0 && (contador == 13 || contador == 16) && ((dezena / 10) == 4))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}